File: js/array.js
class array{
    constructor(){
        this.length=0;
        this.data = {};
    }

    getElementAtIndex(index){
        return this.data[index];
    }

    append(element){
        this.data[this.length]= element
        this.length++;
        return this.length;
    }

    pop(){
        const item = this.data[this.length-1];
        delete this.data[this.length-1];
        this.length--;
        return item;
    }

    deleteAt(index){
        for(let i=index;i<this.length-1;i++){
            this.data[i] = this.data[i+1];
        }
        delete this.data[this.length-1];
        this.length--;
        return this.data
    }

    insertAt(index,item){
        for(let i=this.length;i>=index;i--){
            this.data[i] = this.data[i-1];
        }
        this.data[index] = item;
        this.length++;
        return this.data;
    }
}

array_obj = new array();
array_obj.append(67);
array_obj.append(450);
array_obj.append(560);
array_obj.append(750);
console.log("length of the array is ",array_obj.length)

for(var key in array_obj.data){
    console.log(array_obj.data[key])
}

array_obj.pop()
console.log("After pop")
for(var key in array_obj.data){
    console.log(array_obj.data[key])
}

array_obj.deleteAt(1)

console.log("We have dleted an element at index 1")

for(var key in array_obj.data){
    console.log(array_obj.data[key])
}

array_obj.insertAt(1,500)
console.log("The element 500 is inserted at index 1")

for(var key in array_obj.data){
    console.log(array_obj.data[key])
}

File: linear_data_structure/py/array.py
class array:
    def __init__(self):
        self.length = 0
        self.data = []

    def getElementAtIndex(self,index):
        return self.data[index]

    def append(self,element):
        self.data.append(element)
        self.length+=1
        return self.length
    
    def pop(self):
        item = self.data[self.length-1]
        del self.data[self.length-1]
        self.length-=1
        return item

    def deleteAt(self,index):
        for i in range(index,self.length-1):
            self.data[i] = self.data[i+1]
        del self.data[self.length-1]
        self.length-=1
        return self.data
    
    def insertAt(self,index,item):
        for i in range(self.length,index,-1):
            self.data[i] = self.data[i-1]
        self.data[index] = item
        self.length+=1
        return self.data

array_obj = array()

array_obj.append(23)
array_obj.append(230)
array_obj.append(2300)
array_obj.append(230000)

print(array_obj.data)


array_obj.deleteAt(3)

print("array_obj has deleted an element at index 3")

print(array_obj.data)

print("array_obj[1] = ",array_obj.getElementAtIndex(1))

array_obj.deleteAt(1)

print("After deleting it looks like this",array_obj.data)


